examples/ensemble/plot_stack_predictors.py

```diff
@@ -73,68 +77,68 @@ def load_ames_housing():
 ##############################################################################
 #
 # Before we can use Ames dataset we still need to do some preprocessing.
-# First, the dataset has many missing values. To impute them, we will exchange
-# categorical missing values with the new category 'missing' while the
-# numerical missing values with the 'mean' of the column. We will also encode
-# the categories with either :class:`~sklearn.preprocessing.OneHotEncoder
-# <sklearn.preprocessing.OneHotEncoder>` or
-# :class:`~sklearn.preprocessing.OrdinalEncoder
-# <sklearn.preprocessing.OrdinalEncoder>` depending for which type of model we
-# will use them (linear or non-linear model). To facilitate this preprocessing
-# we will make two pipelines.
-# You can skip this section if your data is ready to use and does
-# not need preprocessing
+# First, we will create the element of the pipeline to automatically select
+# the categorical and numerical columns.
+
+# %%
+from sklearn.compose import make_column_selector
 
+cat_selector = make_column_selector(dtype_include=object)
+num_selector = make_column_selector(dtype_include=np.number)
 
+# %%
+cat_cols = cat_selector(X)
+cat_cols
+
+# %%
+num_cols = num_selector(X)
+num_cols
+
+# %%
+# Then, we will need to design preprocessing pipelines which depends of the
+# ending regressor. If the ending regressor is a linear model, one needs to
+# one-hot encode the categories. If the ending regressor is a tree-based model
+# an ordinal encoder will be sufficient. Besides, numerical values needs to be
+# standardize for a linear model while the raw numerical data can be treated
+# as is by a tree-based model. However, both models are note natively treating
+# missing values.
+#
+# We will first design the pipeline required for the tree-based models.
+
+# %%
 from sklearn.compose import make_column_transformer
 from sklearn.impute import SimpleImputer
 from sklearn.pipeline import make_pipeline
-from sklearn.preprocessing import OneHotEncoder
 from sklearn.preprocessing import OrdinalEncoder
-from sklearn.preprocessing import StandardScaler
-
-
-cat_cols = X.columns[X.dtypes == 'O']
-num_cols = X.columns[X.dtypes == 'float64']
 
-categories = [
-    X[column].unique() for column in X[cat_cols]]
+cat_tree_processor = OrdinalEncoder(
+    handle_unknown="use_encoded_value", unknown_value=-1)
+num_tree_processor = SimpleImputer(strategy="mean", add_indicator=True)
 
-for cat in categories:
-    cat[cat == None] = 'missing'  # noqa
+tree_preprocessor = make_column_transformer(
+    (num_tree_processor, num_cols),
```

Comment:
why not cat_selector and num_selector here?

Reply:
I illustrated l.90 that it gives the columns name and reuse the columns then.

Reply:
fair, but it would also be nice to show off the callable option which I don't think many people know about... I could go either way on this one though.

Reply:
I am okay either way.